src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala

```diff
@@ -1,19 +1,91 @@
 package temple.generate.server.go.service
 
 import temple.generate.server.ServiceRoot
+import temple.generate.server.go.common.GoCommonGenerator._
 import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.utils.StringUtils.doubleQuote
+
+import scala.collection.immutable.ListMap
 
 object GoServiceCommGenerator {
 
-  private[service] def generateImports(root: ServiceRoot): String = mkCode(
-    "import",
-    CodeWrap.parens.tabbed(
-      s""""${root.module}/util"""",
-    ),
-  )
-
-  private[service] def generateStructs(): String = mkCode.lines(
-    "// Handler maintains the list of services and their associated hostnames",
-    s"type Handler struct ${CodeWrap.curly.tabbed("Services map[string]string")}",
-  )
+  private[service] def generateImports(root: ServiceRoot): String =
+    mkCode(
+      "import",
+      CodeWrap.parens.tabbed(
+        doubleQuote("errors"),
+        doubleQuote("fmt"),
+        doubleQuote("net/http"),
+        "",
+        s""""${root.module}/util"""",
+        doubleQuote("github.com/google/uuid"),
+      ),
+    )
+
+  private[service] def generateCommFunctionDecl(root: ServiceRoot, serviceName: String): String =
+    mkCode(
+      genFunctionCall(
+        s"Check${serviceName.capitalize}",
+        s"${serviceName}${root.idAttribute.name.toUpperCase} ${generateGoType(root.idAttribute.attributeType)}",
+        "token string",
+      ),
+      "(bool, error)",
```

Comment:
I wonder if some argument labels would be useful here? Would make it easier to read. Haven't done it elsewhere tho so I guess would be a pain

Reply:
IntelliJ sticks them in, but not so readable on GitHub